dynawo/sources/Models/CPP/ModelNetwork/DYNModelTwoWindingsTransformer.cpp

```diff
@@ -363,18 +367,60 @@ ModelTwoWindingsTransformer::evalYMat() {
 }
 
 void
-ModelTwoWindingsTransformer::init(int& /*yNum*/) {
-  // not needed
+ModelTwoWindingsTransformer::init(int& yNum) {
+  if (!network_->isInitModel()) {
+    yOffset_ = (unsigned int) yNum;
+    unsigned int localIndex = 0;
+
+    if (modelPhaseChanger_) {
+      iSide2YNum_ = localIndex;
+      ++localIndex;
+    }
+
+    yNum += localIndex;
+  }
 }
 
 void
-ModelTwoWindingsTransformer::evalJt(SparseMatrix& /*jt*/, const double& /*cj*/, const int& /*rowOffset*/) {
-  // not needed
+ModelTwoWindingsTransformer::evalJt(SparseMatrix& jt, const double& /*cj*/, const int& rowOffset) {
+  if (network_->isInitModel())
+    return;
+
+  if (modelPhaseChanger_) {
+    int ur1YNum = modelBus1_->urYNum();
+    int ui1YNum = modelBus1_->uiYNum();
+    int ur2YNum = modelBus2_->urYNum();
+    int ui2YNum = modelBus2_->uiYNum();
+
+    double ur1Val = ur1();
+    double ui1Val = ui1();
+    double ur2Val = ur2();
+    double ui2Val = ui2();
+    double ur1Term = (ir2_dUr1() * ir2(ur1Val, ui1Val, ur2Val, ui2Val) + ii2_dUr1() * ii2(ur1Val, ui1Val, ur2Val, ui2Val))
```

Comment:
ir2 and ii2 should be calculate once and for all and stored in local variables to avoid multiple callings

Reply:
done